internal/handlers/handler_register_u2f_step1_test.go

```diff
@@ -0,0 +1,98 @@
+package handlers
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/authelia/authelia/internal/middlewares"
+	"github.com/authelia/authelia/internal/mocks"
+	"github.com/dgrijalva/jwt-go"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type HandlerRegisterU2FStep1Suite struct {
+	suite.Suite
+
+	mock *mocks.MockAutheliaCtx
+}
+
+func (s *HandlerRegisterU2FStep1Suite) SetupTest() {
+	s.mock = mocks.NewMockAutheliaCtx(s.T())
+
+	userSession := s.mock.Ctx.GetSession()
+	userSession.Username = "john"
+	s.mock.Ctx.SaveSession(userSession)
+}
+
+func (s *HandlerRegisterU2FStep1Suite) TearDownTest() {
+	s.mock.Close()
+}
+
+func createToken(secret string, username string, action string, expiresAt time.Time) string {
+	claims := &middlewares.IdentityVerificationClaim{
+		StandardClaims: jwt.StandardClaims{
+			ExpiresAt: expiresAt.Unix(),
+			Issuer:    "Authelia",
+		},
+		Action:   action,
+		Username: username,
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	ss, _ := token.SignedString([]byte(secret))
+	return ss
+}
+
+func newFinishArgs() middlewares.IdentityVerificationFinishArgs {
+	return middlewares.IdentityVerificationFinishArgs{
+		ActionClaim:          U2FRegistrationAction,
+		IsTokenUserValidFunc: func(ctx *middlewares.AutheliaCtx, username string) bool { return true },
+	}
+}
+
+func (s *HandlerRegisterU2FStep1Suite) TestShouldRaiseWhenXForwardedProtoIsMissing() {
+	token := createToken(s.mock.Ctx.Configuration.JWTSecret, "john", U2FRegistrationAction,
+		time.Now().Add(1*time.Minute))
+	s.mock.Ctx.Request.SetBodyString(fmt.Sprintf("{\"token\":\"%s\"}", token))
+
+	s.mock.StorageProviderMock.EXPECT().
+		FindIdentityVerificationToken(gomock.Eq(token)).
+		Return(true, nil)
+
+	s.mock.StorageProviderMock.EXPECT().
+		RemoveIdentityVerificationToken(gomock.Eq(token)).
+		Return(nil)
+
+	SecondFactorU2FIdentityFinish(s.mock.Ctx)
+
+	// Return 200 KO
```

Comment:
Insignificant Typo.

Reply:
Resolving this in favor of the more specific one. There are a few instances of this.